# Default Credit Card

In [1]:
import warnings
warnings.filterwarnings('ignore')

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
dataset = pd.read_excel(r'C:\Users\LENOVO\OneDrive\Desktop\fsds materials\fsds\3. Aug\29th\SVM\project\default of credit card clients.xls')
dataset

ID  LIMIT_BAL  SEX  EDUCATION  MARRIAGE  AGE  PAY_0  PAY_2  PAY_3  \
0          1      20000    2          2         1   24      2      2     -1   
1          2     120000    2          2         2   26     -1      2      0   
2          3      90000    2          2         2   34      0      0      0   
3          4      50000    2          2         1   37      0      0      0   
4          5      50000    1          2         1   57     -1      0     -1   
...      ...        ...  ...        ...       ...  ...    ...    ...    ...   
29995  29996     220000    1          3         1   39      0      0      0   
29996  29997     150000    1          3         2   43     -1     -1     -1   
29997  29998      30000    1          2         2   37      4      3      2   
29998  29999      80000    1          3         1   41      1     -1      0   
29999  30000      50000    1          2         1   46      0      0      0   

       PAY_4  ...  BILL_AMT4  BILL_AMT5  BILL_AMT6  PAY_AMT1  PAY_AMT2  \
0         -1  ...          0          0          0         0       689   
1          0  ...       3272       3455       3261         0      1000   
2          0  ...      14331      14948      15549      1518      1500   
3          0  ...      28314      28959      29547      2000      2019   
4          0  ...      20940      19146      19131      2000     36681   
...      ...  ...        ...        ...        ...       ...       ...   
29995      0  ...      88004      31237      15980      8500     20000   
29996     -1  ...       8979       5190          0      1837      3526   
29997     -1  ...      20878      20582      19357         0         0   
29998      0  ...      52774      11855      48944     85900      3409   
29999      0  ...      36535      32428      15313      2078      1800   

       PAY_AMT3  PAY_AMT4  PAY_AMT5  PAY_AMT6  default payment next month  
0             0         0         0         0                           1  
1          1000      1000         0      2000                           1  
2          1000      1000      1000      5000                           0  
3          1200      1100      1069      1000                           0  
4         10000      9000       689       679                           0  
...         ...       ...       ...       ...                         ...  
29995      5003      3047      5000      1000                           0  
29996      8998       129         0         0                           0  
29997     22000      4200      2000      3100                           1  
29998      1178      1926     52964      1804                           1  
29999      1430      1000      1000      1000                           1  

[30000 rows x 25 columns]

In [4]:
dataset.shape

(30000, 25)

In [5]:
dataset.columns

Index(['ID', 'LIMIT_BAL', 'SEX', 'EDUCATION', 'MARRIAGE', 'AGE', 'PAY_0',
       'PAY_2', 'PAY_3', 'PAY_4', 'PAY_5', 'PAY_6', 'BILL_AMT1', 'BILL_AMT2',
       'BILL_AMT3', 'BILL_AMT4', 'BILL_AMT5', 'BILL_AMT6', 'PAY_AMT1',
       'PAY_AMT2', 'PAY_AMT3', 'PAY_AMT4', 'PAY_AMT5', 'PAY_AMT6',
       'default payment next month'],
      dtype='object')

In [6]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 25 columns):
 #   Column                      Non-Null Count  Dtype
---  ------                      --------------  -----
 0   ID                          30000 non-null  int64
 1   LIMIT_BAL                   30000 non-null  int64
 2   SEX                         30000 non-null  int64
 3   EDUCATION                   30000 non-null  int64
 4   MARRIAGE                    30000 non-null  int64
 5   AGE                         30000 non-null  int64
 6   PAY_0                       30000 non-null  int64
 7   PAY_2                       30000 non-null  int64
 8   PAY_3                       30000 non-null  int64
 9   PAY_4                       30000 non-null  int64
 10  PAY_5                       30000 non-null  int64
 11  PAY_6                       30000 non-null  int64
 12  BILL_AMT1                   30000 non-null  int64
 13  BILL_AMT2                   30000 non-null  int64
 14  BILL_A

In [7]:
dataset.describe()

ID       LIMIT_BAL           SEX     EDUCATION      MARRIAGE  \
count  30000.000000    30000.000000  30000.000000  30000.000000  30000.000000   
mean   15000.500000   167484.322667      1.603733      1.853133      1.551867   
std     8660.398374   129747.661567      0.489129      0.790349      0.521970   
min        1.000000    10000.000000      1.000000      0.000000      0.000000   
25%     7500.750000    50000.000000      1.000000      1.000000      1.000000   
50%    15000.500000   140000.000000      2.000000      2.000000      2.000000   
75%    22500.250000   240000.000000      2.000000      2.000000      2.000000   
max    30000.000000  1000000.000000      2.000000      6.000000      3.000000   

                AGE         PAY_0         PAY_2         PAY_3         PAY_4  \
count  30000.000000  30000.000000  30000.000000  30000.000000  30000.000000   
mean      35.485500     -0.016700     -0.133767     -0.166200     -0.220667   
std        9.217904      1.123802      1.197186      1.196868      1.169139   
min       21.000000     -2.000000     -2.000000     -2.000000     -2.000000   
25%       28.000000     -1.000000     -1.000000     -1.000000     -1.000000   
50%       34.000000      0.000000      0.000000      0.000000      0.000000   
75%       41.000000      0.000000      0.000000      0.000000      0.000000   
max       79.000000      8.000000      8.000000      8.000000      8.000000   

       ...      BILL_AMT4      BILL_AMT5      BILL_AMT6       PAY_AMT1  \
count  ...   30000.000000   30000.000000   30000.000000   30000.000000   
mean   ...   43262.948967   40311.400967   38871.760400    5663.580500   
std    ...   64332.856134   60797.155770   59554.107537   16563.280354   
min    ... -170000.000000  -81334.000000 -339603.000000       0.000000   
25%    ...    2326.750000    1763.000000    1256.000000    1000.000000   
50%    ...   19052.000000   18104.500000   17071.000000    2100.000000   
75%    ...   54506.000000   50190.500000   49198.250000    5006.000000   
max    ...  891586.000000  927171.000000  961664.000000  873552.000000   

           PAY_AMT2      PAY_AMT3       PAY_AMT4       PAY_AMT5  \
count  3.000000e+04   30000.00000   30000.000000   30000.000000   
mean   5.921163e+03    5225.68150    4826.076867    4799.387633   
std    2.304087e+04   17606.96147   15666.159744   15278.305679   
min    0.000000e+00       0.00000       0.000000       0.000000   
25%    8.330000e+02     390.00000     296.000000     252.500000   
50%    2.009000e+03    1800.00000    1500.000000    1500.000000   
75%    5.000000e+03    4505.00000    4013.250000    4031.500000   
max    1.684259e+06  896040.00000  621000.000000  426529.000000   

            PAY_AMT6  default payment next month  
count   30000.000000                30000.000000  
mean     5215.502567                    0.221200  
std     17777.465775                    0.415062  
min         0.000000                    0.000000  
25%       117.750000                    0.000000  
50%      1500.000000                    0.000000  
75%      4000.000000                    0.000000  
max    528666.000000                    1.000000  

[8 rows x 25 columns]

In [8]:
dataset.isnull()

ID  LIMIT_BAL    SEX  EDUCATION  MARRIAGE    AGE  PAY_0  PAY_2  \
0      False      False  False      False     False  False  False  False   
1      False      False  False      False     False  False  False  False   
2      False      False  False      False     False  False  False  False   
3      False      False  False      False     False  False  False  False   
4      False      False  False      False     False  False  False  False   
...      ...        ...    ...        ...       ...    ...    ...    ...   
29995  False      False  False      False     False  False  False  False   
29996  False      False  False      False     False  False  False  False   
29997  False      False  False      False     False  False  False  False   
29998  False      False  False      False     False  False  False  False   
29999  False      False  False      False     False  False  False  False   

       PAY_3  PAY_4  ...  BILL_AMT4  BILL_AMT5  BILL_AMT6  PAY_AMT1  PAY_AMT2  \
0      False  False  ...      False      False      False     False     False   
1      False  False  ...      False      False      False     False     False   
2      False  False  ...      False      False      False     False     False   
3      False  False  ...      False      False      False     False     False   
4      False  False  ...      False      False      False     False     False   
...      ...    ...  ...        ...        ...        ...       ...       ...   
29995  False  False  ...      False      False      False     False     False   
29996  False  False  ...      False      False      False     False     False   
29997  False  False  ...      False      False      False     False     False   
29998  False  False  ...      False      False      False     False     False   
29999  False  False  ...      False      False      False     False     False   

       PAY_AMT3  PAY_AMT4  PAY_AMT5  PAY_AMT6  default payment next month  
0         False     False     False     False                       False  
1         False     False     False     False                       False  
2         False     False     False     False                       False  
3         False     False     False     False                       False  
4         False     False     False     False                       False  
...         ...       ...       ...       ...                         ...  
29995     False     False     False     False                       False  
29996     False     False     False     False                       False  
29997     False     False     False     False                       False  
29998     False     False     False     False                       False  
29999     False     False     False     False                       False  

[30000 rows x 25 columns]

In [9]:
dataset.isnull().sum()

ID                            0
LIMIT_BAL                     0
SEX                           0
EDUCATION                     0
MARRIAGE                      0
AGE                           0
PAY_0                         0
PAY_2                         0
PAY_3                         0
PAY_4                         0
PAY_5                         0
PAY_6                         0
BILL_AMT1                     0
BILL_AMT2                     0
BILL_AMT3                     0
BILL_AMT4                     0
BILL_AMT5                     0
BILL_AMT6                     0
PAY_AMT1                      0
PAY_AMT2                      0
PAY_AMT3                      0
PAY_AMT4                      0
PAY_AMT5                      0
PAY_AMT6                      0
default payment next month    0
dtype: int64

In [10]:
X = dataset.drop(['ID','default payment next month'],axis=1)
X

LIMIT_BAL  SEX  EDUCATION  MARRIAGE  AGE  PAY_0  PAY_2  PAY_3  PAY_4  \
0          20000    2          2         1   24      2      2     -1     -1   
1         120000    2          2         2   26     -1      2      0      0   
2          90000    2          2         2   34      0      0      0      0   
3          50000    2          2         1   37      0      0      0      0   
4          50000    1          2         1   57     -1      0     -1      0   
...          ...  ...        ...       ...  ...    ...    ...    ...    ...   
29995     220000    1          3         1   39      0      0      0      0   
29996     150000    1          3         2   43     -1     -1     -1     -1   
29997      30000    1          2         2   37      4      3      2     -1   
29998      80000    1          3         1   41      1     -1      0      0   
29999      50000    1          2         1   46      0      0      0      0   

       PAY_5  ...  BILL_AMT3  BILL_AMT4  BILL_AMT5  BILL_AMT6  PAY_AMT1  \
0         -2  ...        689          0          0          0         0   
1          0  ...       2682       3272       3455       3261         0   
2          0  ...      13559      14331      14948      15549      1518   
3          0  ...      49291      28314      28959      29547      2000   
4          0  ...      35835      20940      19146      19131      2000   
...      ...  ...        ...        ...        ...        ...       ...   
29995      0  ...     208365      88004      31237      15980      8500   
29996      0  ...       3502       8979       5190          0      1837   
29997      0  ...       2758      20878      20582      19357         0   
29998      0  ...      76304      52774      11855      48944     85900   
29999      0  ...      49764      36535      32428      15313      2078   

       PAY_AMT2  PAY_AMT3  PAY_AMT4  PAY_AMT5  PAY_AMT6  
0           689         0         0         0         0  
1          1000      1000      1000         0      2000  
2          1500      1000      1000      1000      5000  
3          2019      1200      1100      1069      1000  
4         36681     10000      9000       689       679  
...         ...       ...       ...       ...       ...  
29995     20000      5003      3047      5000      1000  
29996      3526      8998       129         0         0  
29997         0     22000      4200      2000      3100  
29998      3409      1178      1926     52964      1804  
29999      1800      1430      1000      1000      1000  

[30000 rows x 23 columns]

In [11]:
y = dataset['default payment next month']
y

0        1
1        1
2        0
3        0
4        0
        ..
29995    0
29996    0
29997    1
29998    1
29999    1
Name: default payment next month, Length: 30000, dtype: int64

In [12]:
y.shape

(30000,)

In [13]:
dataset = pd.get_dummies(dataset,columns=['EDUCATION','MARRIAGE','SEX'],prefix=['EDU',"MRG",('MALE','FEMALE')])

In [14]:
dataset

ID  LIMIT_BAL  AGE  PAY_0  PAY_2  PAY_3  PAY_4  PAY_5  PAY_6  \
0          1      20000   24      2      2     -1     -1     -2     -2   
1          2     120000   26     -1      2      0      0      0      2   
2          3      90000   34      0      0      0      0      0      0   
3          4      50000   37      0      0      0      0      0      0   
4          5      50000   57     -1      0     -1      0      0      0   
...      ...        ...  ...    ...    ...    ...    ...    ...    ...   
29995  29996     220000   39      0      0      0      0      0      0   
29996  29997     150000   43     -1     -1     -1     -1      0      0   
29997  29998      30000   37      4      3      2     -1      0      0   
29998  29999      80000   41      1     -1      0      0      0     -1   
29999  30000      50000   46      0      0      0      0      0      0   

       BILL_AMT1  ...  EDU_3  EDU_4  EDU_5  EDU_6  MRG_0  MRG_1  MRG_2  MRG_3  \
0           3913  ...      0      0      0      0      0      1      0      0   
1           2682  ...      0      0      0      0      0      0      1      0   
2          29239  ...      0      0      0      0      0      0      1      0   
3          46990  ...      0      0      0      0      0      1      0      0   
4           8617  ...      0      0      0      0      0      1      0      0   
...          ...  ...    ...    ...    ...    ...    ...    ...    ...    ...   
29995     188948  ...      1      0      0      0      0      1      0      0   
29996       1683  ...      1      0      0      0      0      0      1      0   
29997       3565  ...      0      0      0      0      0      0      1      0   
29998      -1645  ...      1      0      0      0      0      1      0      0   
29999      47929  ...      0      0      0      0      0      1      0      0   

       ('MALE', 'FEMALE')_1  ('MALE', 'FEMALE')_2  
0                         0                     1  
1                         0                     1  
2                         0                     1  
3                         0                     1  
4                         1                     0  
...                     ...                   ...  
29995                     1                     0  
29996                     1                     0  
29997                     1                     0  
29998                     1                     0  
29999                     1                     0  

[30000 rows x 35 columns]

In [15]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.20,random_state=0)

In [16]:
X_train

LIMIT_BAL  SEX  EDUCATION  MARRIAGE  AGE  PAY_0  PAY_2  PAY_3  PAY_4  \
3225       20000    2          3         1   44      0      0      2      0   
11815     260000    2          2         2   30     -1     -1     -1     -1   
7338       20000    1          2         1   39      2      0      0      0   
14980      30000    1          2         1   23      2      2      2      2   
27167      10000    1          2         1   29      0      0      0      0   
...          ...  ...        ...       ...  ...    ...    ...    ...    ...   
13123      30000    1          2         2   38      0      0      0      0   
19648     210000    2          1         1   33      0      0      0      0   
9845      130000    2          3         1   43      0      0      0      0   
10799      50000    2          3         1   29      0      0      0      0   
2732      140000    2          1         2   25      0      0      0      0   

       PAY_5  ...  BILL_AMT3  BILL_AMT4  BILL_AMT5  BILL_AMT6  PAY_AMT1  \
3225       0  ...      17980      18780          0          0      3000   
11815     -1  ...        274        165        333        165       165   
7338       0  ...      19299      19928      20204      20398      1500   
14980      2  ...      28635      30127      30525      29793      1800   
27167      0  ...       8600       9470       6690       9690      2800   
...      ...  ...        ...        ...        ...        ...       ...   
13123      0  ...      18432      17844      18237      10900      2100   
19648      0  ...     192717     195700     150525     137565      7000   
9845       0  ...      66370      43882      41063      29215      5000   
10799      0  ...      49054       7378       8103       8963      2500   
2732       0  ...      26006      26771      28179      29746      2000   

       PAY_AMT2  PAY_AMT3  PAY_AMT4  PAY_AMT5  PAY_AMT6  
3225          0      1000      1000         0         0  
11815       274       165       333       165       293  
7338       1500       900       700      1480         0  
14980       150      2250      1000         0       700  
27167      2000      1500       900      3000         0  
...         ...       ...       ...       ...       ...  
13123      2500      1000      1000      3000      7002  
19648      7791      7000      5254      5400      5380  
9845       3000      2500      4100         0      5000  
10799      3125       417      1000      1000       500  
2732       2000      1500      2000      2200      2000  

[24000 rows x 23 columns]

In [17]:
X_test

LIMIT_BAL  SEX  EDUCATION  MARRIAGE  AGE  PAY_0  PAY_2  PAY_3  PAY_4  \
8225       20000    1          1         2   33      1      2      2      2   
10794      20000    2          2         2   35      0      0      2      0   
9163      230000    2          1         1   44      1     -1     -1     -1   
26591     100000    1          2         1   42      0      0      0      0   
6631      150000    1          1         2   29     -2     -2     -2     -2   
...          ...  ...        ...       ...  ...    ...    ...    ...    ...   
12715     250000    2          2         1   35      1      2      2      2   
28867     210000    2          1         1   46      0      0      0      0   
3758       80000    1          3         1   27      0      0      0     -1   
17842      20000    1          2         2   24      2      2      2      2   
9119       50000    1          2         2   40      1      1     -1     -1   

       PAY_5  ...  BILL_AMT3  BILL_AMT4  BILL_AMT5  BILL_AMT6  PAY_AMT1  \
8225       2  ...      19057      18453      19755      19288         0   
10794      0  ...      18600      19000      19000      20000      3400   
9163      -1  ...       2864        933          0          0       949   
26591      0  ...      16138      17758      18774      20272      5000   
6631      -2  ...        831       6469       5138       7810      6989   
...      ...  ...        ...        ...        ...        ...       ...   
12715      2  ...     205681     200988     217830     223254         0   
28867      0  ...      48957      49848      59060      60420      3111   
3758       0  ...       7021       6157       7606       3432      1500   
17842      2  ...      16715      18327      17735      19446       600   
9119       0  ...        409      29148      30270      22814         0   

       PAY_AMT2  PAY_AMT3  PAY_AMT4  PAY_AMT5  PAY_AMT6  
8225       2260         0      1600         0       644  
10794         0      1000         0      1000         0  
9163       2873       933         0         0         0  
26591      2000      2000      2000      2000      2000  
6631        833      6488      5153      7833      7130  
...         ...       ...       ...       ...       ...  
12715     17000         0     20000      9078      8500  
28867      2109      2000     10000      6032      5030  
3758       3000      6157      1500       138        59  
17842      1000      2200         0      2000         0  
9119       2931     30000      5270      2814      3089  

[6000 rows x 23 columns]

In [18]:
y_train

3225     0
11815    0
7338     1
14980    1
27167    1
        ..
13123    0
19648    1
9845     0
10799    0
2732     0
Name: default payment next month, Length: 24000, dtype: int64

In [19]:
y_test

8225     0
10794    0
9163     0
26591    0
6631     0
        ..
12715    0
28867    0
3758     1
17842    0
9119     0
Name: default payment next month, Length: 6000, dtype: int64

In [20]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

In [21]:
print("LOGISTIC REGRESSION ALGORITHM")
print()
from sklearn.linear_model import LogisticRegression
lg = LogisticRegression()
lg.fit(X_train,y_train)
y_pred = lg.predict(X_test)
print('y_pred')
print(y_pred)
print()
from sklearn.metrics import confusion_matrix,accuracy_score,classification_report
cm = confusion_matrix(y_test,y_pred)
print('conufusion_matrix')
print()
print(cm)
print()
ac = accuracy_score(y_test,y_pred)
print('accuracy_score')
print(ac)
print()
cr = classification_report(y_test,y_pred)
print('classification_report')
print(cr)
print()
bias = lg.score(X_train,y_train)
print('bias=',bias)
print()
variance = lg.score(X_test,y_test)
print('variance=',variance)

LOGISTIC REGRESSION ALGORITHM

y_pred
[1 0 0 ... 0 1 0]

conufusion_matrix

[[4606   97]
 [ 987  310]]

accuracy_score
0.8193333333333334

classification_report
              precision    recall  f1-score   support

           0       0.82      0.98      0.89      4703
           1       0.76      0.24      0.36      1297

    accuracy                           0.82      6000
   macro avg       0.79      0.61      0.63      6000
weighted avg       0.81      0.82      0.78      6000


bias= 0.808

variance= 0.8193333333333334


In [22]:
print("KNEARESTNEIGHBORS ALGORITHM")
print()
from sklearn.neighbors import KNeighborsClassifier
Knn = KNeighborsClassifier()
Knn.fit(X_train,y_train)
Knn_pred = Knn.predict(X_test)
print('knn_pred')
print(Knn_pred)
print()
from sklearn.metrics import confusion_matrix,accuracy_score,classification_report
cm = confusion_matrix(y_test,y_pred)
print('conufusion_matrix')
print()
print(cm)
print()
ac = accuracy_score(y_test,y_pred)
print('accuracy_score')
print(ac)
print()
cr = classification_report(y_test,y_pred)
print('classification_report')
print(cr)
print()
bias = Knn.score(X_train,y_train)
print('bias=',bias)
print()
variance = Knn.score(X_test,y_test)
print('variance=',variance)


KNEARESTNEIGHBORS ALGORITHM

knn_pred
[0 0 0 ... 0 1 0]

conufusion_matrix

[[4606   97]
 [ 987  310]]

accuracy_score
0.8193333333333334

classification_report
              precision    recall  f1-score   support

           0       0.82      0.98      0.89      4703
           1       0.76      0.24      0.36      1297

    accuracy                           0.82      6000
   macro avg       0.79      0.61      0.63      6000
weighted avg       0.81      0.82      0.78      6000


bias= 0.8431666666666666

variance= 0.7891666666666667


In [23]:
print("SUPPORT VECTOR MACHINE ALGORITHM")
print()
from sklearn.svm import SVC
svc = SVC()
svc.fit(X_train,y_train)
svc_pred = svc.predict(X_test)
print('svc_pred')
print(svc_pred)
print()
from sklearn.metrics import confusion_matrix,accuracy_score,classification_report
cm = confusion_matrix(y_test,y_pred)
print('conufusion_matrix')
print()
print(cm)
print()
ac = accuracy_score(y_test,y_pred)
print('accuracy_score')
print(ac)
print()
cr = classification_report(y_test,y_pred)
print('classification_report')
print(cr)
print()
bias = svc.score(X_train,y_train)
print('bias=',bias)
print()
variance = svc.score(X_test,y_test)
print('variance=',variance)


SUPPORT VECTOR MACHINE ALGORITHM

svc_pred
[1 0 0 ... 0 1 0]

conufusion_matrix

[[4606   97]
 [ 987  310]]

accuracy_score
0.8193333333333334

classification_report
              precision    recall  f1-score   support

           0       0.82      0.98      0.89      4703
           1       0.76      0.24      0.36      1297

    accuracy                           0.82      6000
   macro avg       0.79      0.61      0.63      6000
weighted avg       0.81      0.82      0.78      6000


bias= 0.8236666666666667

variance= 0.826


In [24]:
print("NAVIE BAYES ALGORITHM")
print()
from sklearn.naive_bayes import GaussianNB
gnb = GaussianNB()
gnb.fit(X_train,y_train)
gnb_pred = gnb.predict(X_test)
print('gnb_pred')
print(gnb_pred)
print()
from sklearn.metrics import confusion_matrix,accuracy_score,classification_report
cm = confusion_matrix(y_test,y_pred)
print('conufusion_matrix')
print()
print(cm)
print()
ac = accuracy_score(y_test,y_pred)
print('accuracy_score')
print(ac)
print()
cr = classification_report(y_test,y_pred)
print('classification_report')
print(cr)
print()
bias = gnb.score(X_train,y_train)
print('bias=',bias)
print()
variance = gnb.score(X_test,y_test)
print('variance=',variance)


NAVIE BAYES ALGORITHM

gnb_pred
[1 1 0 ... 0 1 1]

conufusion_matrix

[[4606   97]
 [ 987  310]]

accuracy_score
0.8193333333333334

classification_report
              precision    recall  f1-score   support

           0       0.82      0.98      0.89      4703
           1       0.76      0.24      0.36      1297

    accuracy                           0.82      6000
   macro avg       0.79      0.61      0.63      6000
weighted avg       0.81      0.82      0.78      6000


bias= 0.6879166666666666

variance= 0.6863333333333334


In [25]:
print("NAVIE BAYES ALGORITHM")
print()
from sklearn.naive_bayes import BernoulliNB
bnb = BernoulliNB()
bnb.fit(X_train,y_train)
bnb_pred = bnb.predict(X_test)
print('bnb_pred')
print(bnb_pred)
print()
from sklearn.metrics import confusion_matrix,accuracy_score,classification_report
cm = confusion_matrix(y_test,y_pred)
print('conufusion_matrix')
print()
print(cm)
print()
ac = accuracy_score(y_test,y_pred)
print('accuracy_score')
print(ac)
print()
cr = classification_report(y_test,y_pred)
print('classification_report')
print(cr)
print()
bias = bnb.score(X_train,y_train)
print('bias=',bias)
print()
variance = bnb.score(X_test,y_test)
print('variance=',variance)


NAVIE BAYES ALGORITHM

bnb_pred
[1 1 0 ... 0 1 0]

conufusion_matrix

[[4606   97]
 [ 987  310]]

accuracy_score
0.8193333333333334

classification_report
              precision    recall  f1-score   support

           0       0.82      0.98      0.89      4703
           1       0.76      0.24      0.36      1297

    accuracy                           0.82      6000
   macro avg       0.79      0.61      0.63      6000
weighted avg       0.81      0.82      0.78      6000


bias= 0.671625

variance= 0.6761666666666667


In [26]:
print("DECISION TREE ALGORITHM")
print()
from sklearn.tree import DecisionTreeClassifier
dt = DecisionTreeClassifier()
dt.fit(X_train,y_train)
dt_pred = dt.predict(X_test)
print('gnb_pred')
print(gnb_pred)
print()
from sklearn.metrics import confusion_matrix,accuracy_score,classification_report
cm = confusion_matrix(y_test,y_pred)
print('conufusion_matrix')
print()
print(cm)
print()
ac = accuracy_score(y_test,y_pred)
print('accuracy_score')
print(ac)
print()
cr = classification_report(y_test,y_pred)
print('classification_report')
print(cr)
print()
bias = dt.score(X_train,y_train)
print('bias=',bias)
print()
variance = dt.score(X_test,y_test)
print('variance=',variance)


DECISION TREE ALGORITHM

gnb_pred
[1 1 0 ... 0 1 1]

conufusion_matrix

[[4606   97]
 [ 987  310]]

accuracy_score
0.8193333333333334

classification_report
              precision    recall  f1-score   support

           0       0.82      0.98      0.89      4703
           1       0.76      0.24      0.36      1297

    accuracy                           0.82      6000
   macro avg       0.79      0.61      0.63      6000
weighted avg       0.81      0.82      0.78      6000


bias= 0.999625

variance= 0.7335


# LOGISTIC REGRESSION
## ACCURACY =  0.819
## BIAS =  0.808
## VARIANCE =  0.819

# KNEARESTNEIGHBORS(KNN)
## ACCURACY =   0.819
## BIAS =  0.843
## VARIANCE =  0.789

# SUPPORT VECTOR MACHINE
## ACCURACY =  0.819
## BIAS =   0.823
## VARIANCE =  0.826

# NAVIE-BAYES (BERNOULLI)
## ACCURACY =  0.819
## BIAS =   0.671
## VARIANCE =  0.676

# NAVIE-BAYES (GAUSSIAN)
## ACCURACY =  0.819
## BIAS =   0.87
## VARIANCE =  0.86

# DECISION TREE
## ACCURACY =  0.819
## BIAS =   0.99
## VARIANCE =  0.733